exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/project/ProjectRecordBatch.java

```diff
@@ -122,6 +122,7 @@ protected void killIncoming(final boolean sendUpstream) {
 
   @Override
   public IterOutcome innerNext() {
+    recordCount = 0;
```

Comment:
Why don't you add this to parent class rather than each child of the parent? (Either AbstractRecordBatch or AbstractSingleRecordBatch.)